khmer/_oxli/graphs.pyx

```diff
@@ -1,20 +1,261 @@
+        """Get the larget count of all the k-mers in the string."""
+        data = self._valid_sequence(sequence)
+        return deref(self.c_table).get_max_count(data)
+
+    def get_median_count(self, sequence):
+        """median, average, and stddev of the k-mer counts in sequence."""
+        data = self._valid_sequence(sequence)
+        cdef BoundedCounterType med = 0
+        cdef float average = 0
+        cdef float stddev = 0
+
+        deref(self.c_table).get_median_count(data, med, average, stddev)
+        return (med, average, stddev)
+
+    def get_kmer_hashes(self, sequence):
+        """Retrieve hashes of all k-mers in sequence.
+
+        Hashes are returned in the same order as k-mers appear in sequence.
+        """
+        data = self._valid_sequence(sequence)
+        cdef vector[HashIntoType] hashes
+        deref(self.c_table).get_kmer_hashes(data, hashes)
+        return hashes
+
+    def trim_on_abundance(self, sequence, abundance):
+        """Trim sequence at first k-mer below the given abundance."""
+        data = self._valid_sequence(sequence)
+        trimmed_at = deref(self.c_table).trim_on_abundance(data, abundance)
+        return sequence[:trimmed_at], trimmed_at
+
+    def trim_below_abundance(self, sequence, abundance):
+        """Trim sequence at first k-mer above the given abundance."""
+        data = self._valid_sequence(sequence)
+        trimmed_at = deref(self.c_table).trim_below_abundance(data, abundance)
+        return sequence[:trimmed_at], trimmed_at
+
+    def find_spectral_error_positions(self, sequence, max_count):
+        """Identify positions of low-abundance k-mers."""
+        data = self._valid_sequence(sequence)
+        posns = (deref(self.c_table).find_spectral_error_positions(data,
+                                                                   max_count))
+        return posns
+
+    def consume_seqfile_with_reads_parser(self, read_parser):
+       """Count all k-mers from read_parser."""
+       cdef unsigned long long n_consumed = 0
+       cdef unsigned int total_reads = 0
+
+       cdef CPyReadParser_Object* parser = <CPyReadParser_Object*>read_parser
+
+       deref(self.c_table).consume_seqfile[CpFastxReader](parser.parser,
+                                                          total_reads,
+                                                          n_consumed)
+
+    def consume_seqfile(self, file_name):
+        """Count all k-mers from file_name."""
+        cdef unsigned long long n_consumed = 0
+        cdef unsigned int total_reads = 0
+
+        read_parser = ReadParser(file_name)
+        cdef CPyReadParser_Object* parser = <CPyReadParser_Object*>read_parser
+        deref(self.c_table).consume_seqfile[CpFastxReader](parser.parser,
+                                                           total_reads,
+                                                           n_consumed)
+
+    def abundance_distribution(self, file_name, tracking):
```

Comment:
Why not require this to be `Hashtable tracking`? The CPython version gets deprecated via this PR anyhow.